sparse/local.go

```diff
@@ -0,0 +1,172 @@
+package sparse
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// SyncLocalFile syncs a local file from sourceFilePath to targetFilePath.
+// It ensures the file size is a multiple of the Block constant before syncing.
+func SyncLocalFile(sourceFilePath, targetFilePath string) error {
+	log := logrus.WithFields(logrus.Fields{
+		"sourceFilePath": sourceFilePath,
+		"targetFilePath": targetFilePath,
+	})
+
+	fileInfo, err := os.Stat(sourceFilePath)
+	if err != nil {
+		log.WithError(err).Error("Failed to get file info of source file")
+		return err
+	}
+
+	fileSize := fileInfo.Size()
+	if fileSize%Blocks != 0 {
```

Comment:
nit: `Blocks` is the block size here and a bit weird. We can rename it to block size in the future.